### PR TITLE
Fix broken anchors links

### PR DESCRIPTION
**Pull request by yehor-manzhula**

### CODE REVIEW

File: ilc/client/TransitionManager/TransitionManager.js [hunk 1]
```diff
@@ -227,37 +245,41 @@ export class TransactionManager {
         this.#spinnerTimeout = null;
     };
 
+    #addTransactionBlocker = (transactionBlocker) => {
+        transactionBlocker.finally(() => this.#removeTransactionBlocker(transactionBlocker.getId()));
+        this.#transactionBlockers.add(transactionBlocker);
+    }
+
     #transactionBlockerExists = (blockerId) => {
```
Comment (b1ff) [inline ilc/client/TransitionManager/TransitionManager.js:253]: probably methods also require renaming

File: ilc/client/TransitionManager/TransitionManager.js [hunk 2]
```diff
@@ -92,48 +90,51 @@ export class TransactionManager {
...
+
+                resolve();
+            });
+
+            const targetNode = getSlotElement(slotName);
+            targetNode.style.display = 'none'; // we will show all new slots, only when all will be settled
+            this.#hiddenSlots.push(targetNode);
+            observer.observe(targetNode, { childList: true, subtree: true, attributeFilter: ['style'] });
+        })
+        .onDestroy(() => observer.disconnect());
```
Comment (b1ff) [inline ilc/client/TransitionManager/TransitionManager.js:135]: observer?.disconnect() ?

**Review by b1ff — CHANGES_REQUESTED** [review]

File: ilc/client/TransitionManager/TransitionManager.js (hunk 1 above)
Comment (yehor-manzhula) [reply ilc/client/TransitionManager/TransitionManager.js]: renamed

File: ilc/client/TransitionManager/TransitionManager.js (hunk 2 above)
Comment (yehor-manzhula) [reply ilc/client/TransitionManager/TransitionManager.js]: added condition

**Review by b1ff — APPROVED** [review]

File: ilc/client/TransitionManager/TransitionBlocker.js [hunk 3]
```diff
@@ -0,0 +1,47 @@
+
+export default class TransitionBlocker {
```
Comment (blackrabbit99) [inline ilc/client/TransitionManager/TransitionBlocker.js:2]: What is the responsibility of its class ?

File: ilc/client/TransitionManager/TransitionManager.js [hunk 4]
```diff
@@ -164,22 +169,39 @@ export class TransactionManager {
...
+        
         this.#removeGlobalSpinner();
         document.body.removeAttribute('name');
-        scrollRestorer.restoreScroll(window.history.state ? window.history : {state: {scroll: {x: 0, y: 0}}});
+
+        if (window.location.hash) {
+            const anchorElement = document.querySelector(window.location.hash);
+            if (anchorElement) {
+                anchorElement.scrollIntoView();
+            }
```
Comment (blackrabbit99) [inline ilc/client/TransitionManager/TransitionManager.js:180]: else window.scroll(0, 0); ?

File: ilc/client/TransitionManager/TransitionManager.js [hunk 5]
```diff
@@ -284,12 +308,12 @@ export class TransactionManager {
 
 let defaultInstance = null;
 /**
- * @return {TransactionManager}
+ * @return {TransitionManager}
  */
 export default function defaultFactory() {
     if (defaultInstance === null) {
```
Comment (blackrabbit99) [inline ilc/client/TransitionManager/TransitionManager.js:314]: can we rename it to understand `defaultInstance` of what ?

File: ilc/client/TransitionManager/TransitionManager.js [hunk 6]
```diff
@@ -84,56 +82,63 @@ export class TransactionManager {
...
+        const contentListenerBlocker = new NamedTransactionBlocker(slotName, (resolve) => {
+            this.#runGlobalSpinner();
 
-            // if we have rendered MS to DOM but meaningful content isn't rendered, e.g. due to essential data preload
-            if (!status.hasTextOrOpticNodes) {
-                const hasText = !!targetNode.innerText.trim().length
-                const hasOpticNodes = !!targetNode.querySelector(':not(div):not(span)');
-                status.hasTextOrOpticNodes = hasText || hasOpticNodes;
+            if (window.location.hash) {
+                document.body.setAttribute('name', window.location.hash.slice(1));
```
[the remark this answers is not in the document]
Comment (yehor-manzhula) [reply ilc/client/TransitionManager/TransitionManager.js:99]: As I can see from the tests, it is used for testing purposes to check if page transition has happened.

**Review by blackrabbit99 — APPROVED** [review]

**Pull request merged**